src/mpeg4/appleTag.ts

```diff
@@ -0,0 +1,1314 @@
+     * Gets and sets the Publisher
+     */
+    public get publisher(): string {
+        return this.getDashBox("com.apple.iTunes", "publisher");
+    }
+    public set publisher(v: string) {
+        this.setDashBox("com.apple.iTunes", "publisher", v);
+    }
+
+    /**
+     * Gets and sets the Remixer
+     */
+    public get remixedBy(): string {
+        return this.getDashBox("com.apple.iTunes", "REMIXEDBY");
+    }
+    public set remixedBy(v: string) {
+        this.setDashBox("com.apple.iTunes", "REMIXEDBY", v);
+    }
+
+    /**
+     * Gets and sets a collection of pictures associated with the media represented by the current instance.
+     */
+    public get pictures(): IPicture[] {
+        const l: IPicture[] = [];
+
+        for (const box of this.dataBoxesFromTypeParams(Mpeg4BoxType.Covr)) {
+            const p: IPicture = Picture.fromData(box.data);
+            l.push(p);
+        }
+
+        return l;
+    }
+    public set pictures(v: IPicture[]) {
+        if (v === null || v.length === 0) {
+            this.clearData(Mpeg4BoxType.Covr);
+            return;
+        }
+
+        const boxes: AppleDataBox[] = [];
+
+        for (const picture of v) {
+            let type: AppleDataBoxFlagType = AppleDataBoxFlagType.ContainsData;
+
+            if (picture.mimeType === "image/jpeg") {
+                type = AppleDataBoxFlagType.ContainsJpegData;
+            } else if (picture.mimeType === "image/png") {
+                type = AppleDataBoxFlagType.ContainsPngData;
+            } else if (picture.mimeType === "image/x-windows-bmp") {
+                type = AppleDataBoxFlagType.ContainsBmpData;
+            }
+
+            boxes.push(AppleDataBox.fromDataAndFlags(picture.data, type));
+        }
+
+        this.setDataFromTypeAndBoxes(Mpeg4BoxType.Covr, boxes);
+    }
+
+    /**
+     * Gets whether or not the current instance is empty.
+     */
+    public get isEmpty(): boolean {
+        return !this._ilstBox.hasChildren;
+    }
+
+    /**
+     * Clears the values stored in the current instance.
+     */
+    public clear(): void {
+        this._ilstBox.clearChildren();
+    }
+
+    /**
+     * Gets all data boxes that match any of the provided types.
+     * @param types A @see ByteVector[] object enumerating a list of box types to match.
+     * @returns A @see AppleDataBox[] object enumerating the matching boxes.
+     */
+    public dataBoxesFromTypes(types: ByteVector[]): AppleDataBox[] {
+        const dataBoxes: AppleDataBox[] = [];
+
+        /**
+         * Check each box to see if the match any of the provided types.
+         * If a match is found, loop through the children and add any data box.
+         */
+        for (const box of this._ilstBox.children) {
+            for (const byteVector of types) {
+                if (!ByteVector.equals(Mpeg4Utils.fixId(byteVector), box.boxType)) {
+                    continue;
+                }
+
+                for (const dataBox of box.children) {

```

Comment:
Iv'e extracted the code that gets AppleDataBoxes from a collection of Mpeg4Box to a separate function and changed the way of returning to better match the original implementation. Well, they're using yield return. I've tried to match that. See 3a9095f7caf947374ca32ae2820542b449711ad5.

Reply:
I think this change might've changed the behavior of the methods, though. Looks like the original implementation would collect the apple data boxes from *all* ilst children of the given type*s*. The updated implementation would only collect apple data boxes from the first ilst children of the first given type. Is that a safe change?

What I was originally thinking with the comment was that there's a handful of places where the children of a box are iterated over to find objects of a specific type. I was thinking maybe adding a method onto `Mpeg4Box` like `getChildren<TBox>` that does the iterating over the children and returning a `TBox[]` of the ones that are `instanceof TBox`. Could even be a one lines with `children.filter(b => b instanceof TBox)`, I think 😄 Thoughts?

Reply:
Indeed. This change was caused by my limited understanding of C#'s yield. I've read some more about yield tonight (although I admit it remains somewhat mysterious to me) and I see the problem with my change. I've reverted the change in 439d60d7b29fad4a1e0297f42155fb10ea35e230.
As for using a generic function, that doesn't seem so easy in Typescript. The type check using `instanceof` expects a value and not a type:

![image](https://github.com/benrr101/node-taglib-sharp/assets/11945339/7b444571-3f8b-44e6-8ae9-1bada4e76972)

![image](https://github.com/benrr101/node-taglib-sharp/assets/11945339/825e52e7-ce31-4c2e-90f0-bce873ee284f)

If found some info here about why this is happening: https://stackoverflow.com/questions/46703364/why-does-instanceof-in-typescript-give-me-the-error-foo-only-refers-to-a-ty

Reply:
Haha, I guess I've been living in C# land for too long and forgot the differences between generics in typescript and C#. You're right, because `TBox` is a type, it'll be erased when it's transpiled to javascript. I encountered a similar issue in ID3v2 when getting frames of a certain type. If you're curious on how I implemented that, take a look at https://github.com/benrr101/node-taglib-sharp/blob/48e2338ce4395dd6a1a23efa5fdf1e38cf069896/src/id3v2/id3v2Tag.ts#L1077 But don't feel obligated to implement a whole boxClassType sort of thing here. If we're using this code pattern 3 or more times, it might be worth it, but otherwise, it's not a huge deal.

Reply:
I've attempted something similar in 5f2d33aba6e6bb285a03e5da7b1eae7cff7fadef based on what you did for frames. Let me know what you think.

Reply:
Love it! It's perfect!